cli/azd/cmd/provision.go

```diff
@@ -79,6 +80,7 @@ type provisionAction struct {
 	projectConfig    *project.ProjectConfig
 	writer           io.Writer
 	console          input.Console
+	subService       *account.SubscriptionsService
```

Comment:
Change this for a `SubscriptionsManager` 

Then use `GetSubscriptions(ctx context.Context)` ...  (this will pull the subs from cache) and loop the output to find the subId (or use slices.IndexFunc()).  

For the `location`, you can use `GetLocation()` from the env

resurce group must be stored within the env if we want to display it on future operations..  You can do this as a follow up PR

Reply:
I saw that the Resource group appears in all the tests I did so, I didn't include it to avoid repetition.
but I will test it more and see if it doesn't appear and if that's the case I will add it.

I didn't use the Subscription Manager as it checks for the tenant before getting the name which is not necessary as we already have the tenant ID as an environment variable but I didn't know that SubscriptionsManager gets the subs from the cache, that's way much better thanks!

Reply:
done